be/src/exprs/map_expr.cpp

```diff
@@ -22,19 +23,66 @@
 namespace starrocks {
 
 StatusOr<ColumnPtr> MapExpr::evaluate_checked(ExprContext* context, Chunk* chunk) {
-    DCHECK_EQ(2, _children.size());
-    ASSIGN_OR_RETURN(ColumnPtr key_col, _children[0]->evaluate_checked(context, chunk));
-    ASSIGN_OR_RETURN(ColumnPtr value_col, _children[1]->evaluate_checked(context, chunk));
-    size_t num_rows = std::max(key_col->size(), value_col->size());
-    // No optimization for const column now.
-    key_col = ColumnHelper::unfold_const_column(_children[0]->type(), num_rows, key_col);
-    value_col = ColumnHelper::unfold_const_column(_children[1]->type(), num_rows, value_col);
-    key_col = ColumnHelper::cast_to_nullable_column(key_col);
-    value_col = ColumnHelper::cast_to_nullable_column(value_col);
-    // fake offsets, just for creating map column.
+    if (UNLIKELY(_children.size() % 2 == 1)) {
+        return Status::RuntimeError(fmt::format("Map expressions' keys and values should be in pair."));
+    }
```

Comment:
DCHECK? it's must right in normal env?

Reply:
the inputs should be in pair, otherwise, lossing a value in key-value pairs.